plugins/search/src/components/SearchModal/useSearchModal.tsx

```diff
@@ -135,8 +139,32 @@ export function useSearchModal(initialState = false) {
     [],
   );
 
+  // Check for any existing parent context.
+  const parentContext = useContext(SearchModalContext);
+  const parentContextValue = parentContext?.atVersion(1);
+  const isParentContextPresent = !!parentContextValue?.state;
+
+  // Monitor route pathname changes to automatically hide the modal.
+  const { pathname } = useLocation();
+  const [openedPathname, setOpenedPathname] = useState<string | null>(null);
+  const { open, hidden } = state;
+  const isModalOpened = !isParentContextPresent && open && !hidden;
+  useEffect(() => {
+    if (isModalOpened) {
+      setOpenedPathname(pathname);
```

Comment:
the amount of `if` checking and logic expressions around these two effects makes logic a bit hard to track - for example, I'm not sure that this setter won't accidentally be run as a side effect of locations changing, ie when the navigation event happens, will this _overwrite_ the pathname that was in there before? Does it need a `openedPathname === null` check and also resetting it in the effect below? I'm honestly not sure.

Reply:
We've had a little poke at this (with the similar removal of the toggle) and managed to get what we think is a working solution with a single `useEffect` that looks like this... 

```ts
useEffect(() => {
    setState(prevState => prevState.hidden ? prevState : { ...prevState, hidden: true });
  }, [location.pathname]);
```

but we're also wondering if we're missing an edge case that needs the extra logic you've added?

Reply:
Yeah! That's the kind of fix I'd been hoping would be viable! Both clever and also slightly hacky to rely in `hidden` to always be true on first paint though :P It'll probably always be true except perhaps in tests, so could be OK. Otherwise just some minor adjustment to ignore the effect on the very first render.

Maybe the most robust would be

```ts
import usePrevious from 'react-use/lib/usePrevious';
const previousPathname = usePrevious(location.pathname);
useEffect(() => {
  if (location.pathname !== previousPathname) {
    setState(prevState => { ...prevState, hidden: true });
  }
}, [previousPathname, location.pathname]);
```

Reply:
(the above works if usePrevious returns undefined until the value changes; haven't double checked that yet

Reply:
@freben, @HHogg I tested your suggestions and added a new commit simplifying the solution.